### ISSUE TITLE
Failure reading liquidity_pool_id from trades

### ISSUE THREAD

**Issue by vinamogit** [stellar/java-stellar-sdk#421]
### What version are you using?

java-stellar-sdk 0.32.0

### What did you do?

Trying to get trades for liquidity pools:
https://gist.github.com/vinamogit/312fa993e449d3c57f4db0f7ec52c3a2

This gist also contains a potential fix, not sure this is the way you want to fix it.

### What did you expect to see?

A result without error.

### What did you see instead?

```

Exception in thread "main" shadow.com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at path $.records[1].base_liquidity_pool_id
	at shadow.com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:226)
	at shadow.com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131)
	at shadow.com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222)
	at shadow.com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41)
	at shadow.com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:82)
	at shadow.com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:61)
	at shadow.com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131)
	at shadow.com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222)
	at shadow.com.google.gson.Gson.fromJson(Gson.java:927)
	at shadow.com.google.gson.Gson.fromJson(Gson.java:994)
	at org.stellar.sdk.responses.PageDeserializer.deserialize(PageDeserializer.java:50)
	at org.stellar.sdk.responses.PageDeserializer.deserialize(PageDeserializer.java:19)
	at shadow.com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69)
	at shadow.com.google.gson.Gson.fromJson(Gson.java:927)
	at shadow.com.google.gson.Gson.fromJson(Gson.java:892)
	at shadow.com.google.gson.Gson.fromJson(Gson.java:841)
	at org.stellar.sdk.requests.ResponseHandler.handleResponse(ResponseHandler.java:42)
	at org.stellar.sdk.requests.TradesRequestBuilder.execute(TradesRequestBuilder.java:100)
	at org.stellar.sdk.requests.TradesRequestBuilder.execute(TradesRequestBuilder.java:104)
	at arbitr.Test.main(Test.java:17)
Caused by: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at path $.records[1].base_liquidity_pool_id
	at shadow.com.google.gson.internal.bind.JsonTreeReader.expect(JsonTreeReader.java:162)
	at shadow.com.google.gson.internal.bind.JsonTreeReader.beginObject(JsonTreeReader.java:87)
	at shadow.com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:215)
	... 19 more

```

**Comment by vinamogit** [comment]
This might also fix issue #404.
I could do a PR if you want.

**Comment by overcat** [comment]
Fixed in #422
